pkg/dao/dao.go

```diff
@@ -105,26 +111,22 @@ func (d *Dao) BatchGetRaw(dir string) (*[]string, error) {
 	return &payloads, nil
 }
 
+//GetSpec - Retrieve the spec for the kvp API.
 func (d *Dao) GetSpec(id string) (*apb.Spec, error) {
-	raw, err := d.GetRaw(specKey(id))
+	spec := &apb.Spec{}
+	err := d.getObject(specKey(id), spec)
```

Comment:
Let's also change it to include the error check in the same line.
```
if err := d.getObject(specKey(id), spec); err != nil {
   return nil, err
}
```
I know it will result in more changes, but I think it looks better than what we had.